### ISSUE TITLE
feat: add possibility to render enums as radios

### ISSUE THREAD

**Issue by janhesters** [SeasonedSoftware/remix-forms#54]
# Feature Request

## Current Behavior

Enums are always rendered as `<Select />` components.

## Desired Behavior 

There should be a way to configure `enum`s to be rendered as radio components as well as how the radio group should be rendered, instead of rendering them as `select`s.

For example:

```tsx
import { forwardRef } from 'react';
import type { FormProps } from 'remix-forms';
import { Form as RemixForm } from 'remix-forms';
import type { SomeZodObject } from 'zod';

const Radio = forwardRef<HTMLInputElement, JSX.IntrinsicElements['input']>(
  ({ type = 'radio', ...props }, reference) => (
    <input
      className="focus:ring-indigo-500 h-4 w-4 text-indigo-600 border-gray-300"
      ref={reference}
      type={type}
      {...props}
    />
  ),
);

const Select = forwardRef<HTMLSelectElement, JSX.IntrinsicElements['select']>(
  (props, reference) => (
    <select
      className="shadow-sm focus:ring-indigo-500 focus:border-indigo-500 block w-full sm:text-sm border-gray-300 rounded-md"
      ref={reference}
      {...props}
    />
  ),
);

const Form = <Schema extends SomeZodObject>(props: FormProps<Schema>) => (
  <RemixForm<Schema>
    className="space-y-8 divide-y divide-gray-200"
    radioComponent={Radio}
    selectComponent={Select}
    renderRadioGroup={({ Radio, Label, options }) => (
      <div className="mt-4 space-y-4">
        {options.map(({ name, value }) => (
          <div className="flex items-center">
            <Radio value={value} />
            <Label>{name}</Label>
          </div>
        )}
      </div>
    )}
    {...props}
  />
);
```

**Note:** We have to flush out the details here. In a real form with radio inputs, you usually have labels and the inputs linked using `id`s and `htmlFor` tags.

And then you can use it like this:

```tsx
import Form from '~/where-your-form-lives';
import { z } from 'zod';

const schema = z.object({
  country: z.enum(['United States', 'Canada', 'Mexico']),
  pushNotifications: z.enum([
    'Everything',
    'Same as email',
    'No push notifications',
  ]),
});

export default MyPage() {
  return (
    <Form radio=[['pushNotifications']] schema={schema}>
      {({ Field, Button }) => (
        <>
          <Field name="country" />
          <Field name="pushNotifications" />
          <Button />
        </>
      )
    </Form>
  )
}
```

## Alternatives Considered

You can hack your input fields today by using `register` and rendering your own `Radio`s and `Label`s - like this:

```tsx
<fieldset className="mt-6">
  <legend className="contents text-base font-medium text-gray-900">
    Push Notifications
  </legend>

  <p className="text-sm text-gray-500">
    These are delivered via SMS to your mobile phone.
  </p>

  <Field
    className="mt-4 space-y-4"
    name="pushNotifications"
  >
    {({ Errors }) => (
      <>
        <div className="flex items-center">
          <Radio
            {...register('pushNotifications')}
            checked
            id="push-everything"
            value="Everything"
          />

          <Label
            className="ml-3 block text-sm font-medium text-gray-700"
            htmlFor="push-everything"
          >
            Everything
          </Label>
        </div>

        <div className="flex items-center">
          <Radio
            {...register('pushNotifications')}
            id="push-email"
            value="Same as email"
          />

          <Label
            className="ml-3 block text-sm font-medium text-gray-700"
            htmlFor="push-email"
          >
            Same as email
          </Label>
        </div>

        <div className="flex items-center">
          <Radio
            {...register('pushNotifications')}
            id="push-nothing"
            value="No push notifications"
          />

          <Label
            className="ml-3 block text-sm font-medium text-gray-700"
            htmlFor="push-nothing"
          >
            No push notifications
          </Label>
        </div>

        <Errors />
      </>
    )}
  </Field>
</fieldset>
```

However, that is very tedious and takes a lot of code.

**Comment by xiaoyu-tamu** [comment]
Would be nice remix forms can support radio

**Comment by diogob** [comment]
closed with the release of [1.2.0](https://github.com/SeasonedSoftware/remix-forms/releases/tag/v1.2.0)
